src/tests/e2etest/CommandRunnerTests.cs

```diff
@@ -1,19 +1,279 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using Microsoft.Azure.Devices;
+using Microsoft.Azure.Devices.Shared;
 using NUnit.Framework;
 using System;
 using System.Text.Json;
 using System.Threading.Tasks;
 
 namespace E2eTesting
 {
-    public class CommandRunnerTests : E2eTest
+
+    [TestFixture]
+    public abstract class E2ETest
     {
-        private const string componentName = "CommandRunner";
-        private const string commandArguments = "CommandArguments";
-        private const string commandStatus = "CommandStatus";
+        private RegistryManager _registryManager;
+
+        private readonly string _iotHubConnectionString = Environment.GetEnvironmentVariable("E2E_OSCONFIG_IOTHUB_CONNSTR")?.Trim('"');
+        private readonly string _moduleId = "osconfig";
+        private readonly string _deviceId = Environment.GetEnvironmentVariable("E2E_OSCONFIG_DEVICE_ID");
+
+        private readonly string _sasToken = Environment.GetEnvironmentVariable("E2E_OSCONFIG_SAS_TOKEN");
+        private readonly string _uploadUrl = Environment.GetEnvironmentVariable("E2E_OSCONFIG_UPLOAD_URL")?.Trim('\"');
+        private readonly string _resourceGroupName = Environment.GetEnvironmentVariable("E2E_OSCONFIG_RESOURCE_GROUP_NAME")?.Trim('\"');
+
+        public partial class GenericResponse<T>
+        {
+            public T value { get; set; }
+            public int ac { get; set; }
+        }
+
+        [OneTimeSetUp]
+        public void OneTimeSetUp()
+        {
+            _registryManager = RegistryManager.CreateFromConnectionString(_iotHubConnectionString);
+
+            if ((null == _sasToken) || (null == _uploadUrl) || (null == _resourceGroupName))
+            {
+                Assert.Warn("Missing environment vars required for log upload to blob store");
+            }
+        }
+
+        [OneTimeTearDown]
+        public void OneTimeTearDown()
+        {
+            if ((null == _sasToken) || (null == _uploadUrl) || (null == _resourceGroupName))
+            {
+                Console.WriteLine("[TearDown] Uploading logs to blob store");
+
+                string fileName = String.Format("{0}-{1}.tar.gz", _resourceGroupName, _deviceId);
+                string fullURL = String.Format("{0}{1}?{2}", _uploadUrl, fileName, _sasToken);
+
+                string tempFileCommand = "temp_file=`sudo mktemp`";
+                string tarCommand = "sudo tar -cvzf $temp_file osconfig*.log";
+                string curlCommand = String.Format("curl -X PUT -T $temp_file -H \"x-ms-date: $(date -u)\" -H \"x-ms-blob-type: BlockBlob\" \"{1}\"", fileName, fullURL);
+                string uploadCommand = String.Format("cd /var/log && {0} && {1} && {2}", tempFileCommand, tarCommand, curlCommand);
+
+                // Temporarily commented out to avoid uploading logs to blob store twice
```

Comment:
Im confused about where the duplicate happens? How are the logs now uploaded to blob store?

Reply:
Duplicate would be due to the separate base classes for CommandRunner and all other tests. I am moving forward with using this refactored class for all tests and there will be a single log upload from `E2ETest.cs`.